cogs/reward.py

```diff
@@ -123,40 +108,29 @@ def convert_to_seconds(s):
             resultLevel = charDict["Level"]
             resultCP = charDict["CP"] + treasureArray[0]
             
-            if tierNum == 0 and resultCP > 4:   # For when levels are gained in tier 1
-                while resultLevel < 5:
-                    resultLevel += 1
-                    resultCP -= 4
-                    if resultCP < 5: # break the loop if they wouldn't level past 4
-                        nextCP = resultCP
-                        break
-                    if resultLevel == 5: #if they go outside of tier 1
-                        resultLevel += resultCP // 10
-                        nextCP = resultCP % 10
-            elif tierNum == 0:
-                nextCP = resultCP
+            # CP and level calculations
+            if resultLevel < 5:
+                maxCP = 4
             else:
-                resultLevel += resultCP // 10
-                nextCP = resultCP % 10
-
-            if resultLevel > 20:    # If tier 5 is reached, level is capped at 20
-                levelDifference = resultLevel - 20
-                nextCP += (levelDifference * 10)
-                resultLevel = 20
+                maxCP = 10
                 
+            while(resultCP >= maxCP and resultLevel <20):
+                resultCP -= maxCP
+                resultLevel += 1
+                if resultLevel > 4:
+                    maxCP = 10
+            
+            # A for loop that prints out multiple tiers of TP
+            tpString = ""
+            for x in treasureArray[1]:
+                #print(x)
+                tpString += str(treasureArray[1].get(x)) + " "
+                tpString += x + ", "
+                print(tpString)
```

Comment:
The entire loop can be replaced with this line.
`tpString = ", ".join([f"{value} {key}" for key, value in treasureArray[1].items()])+","`

The solution is fine, but since you don't have much python experience I just want to show a more python-esque method.

Reply:
Up to you if you feel it should be replaced.